### PR TITLE
Created list contacts UI

### LINKED ISSUE #4: Create list contacts UI

### PR DESCRIPTION
**Pull request by adil62**
Fixes #4 

@yedhink _a

Screenshot 2022-09-22 at 9 19 26 PM

Screenshot 2022-09-22 at 9 19 47 PM

### CODE REVIEW

File: app/javascript/src/components/Dashboard/Contacts/DeleteAlert.jsx [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+      logger.error(error);
+      setDeleting(false);
+    }
+  };
+
+  return (
+    <Alert
+      isOpen
+      isSubmitting={deleting}
+      message="Are you sure you want to delete the contact? This action cannot be undone."
```
Comment (yedhink) [inline app/javascript/src/components/Dashboard/Contacts/DeleteAlert.jsx:34]:
```suggestion
      message="Are you sure you want to delete this contact? This action cannot be undone."
```

File: app/javascript/src/components/Dashboard/Contacts/Menubar.jsx [hunk 2]
```diff
@@ -0,0 +1,66 @@
+import React, { useState } from "react";
+
+import { Search, Settings, Plus } from "neetoicons";
+import { Typography } from "neetoui";
+import { MenuBar as NeetoUIMenuBar } from "neetoui/layouts";
+
+const MenuBar = ({ showMenu = false }) => {
```
Comment (yedhink) [inline app/javascript/src/components/Dashboard/Contacts/Menubar.jsx:7]: The filename and the component name doesn't match.

File: app/javascript/src/components/Dashboard/Contacts/Table.jsx [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+      <Typography style="body3">{role}</Typography>
+    </div>
+  </div>
+);
+
+const renderActionDropdown = (id, handleDelete) => (
+  <ActionDropdown
+    entityId={id}
+    handleDelete={handleDelete}
+    handleEdit={noop}
```
Comment (yedhink) [inline app/javascript/src/components/Dashboard/Contacts/Table.jsx:29]: Nice. 👍

File: app/javascript/src/components/Dashboard/Contacts/Table.jsx [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+            },
+            {
+              title: "EMAIL",
+              dataIndex: "email",
+              key: "email",
+            },
+            {
+              title: "CREATED AT",
+              dataIndex: "email",
+              key: "email",
```
Comment (yedhink) [inline app/javascript/src/components/Dashboard/Contacts/Table.jsx:58]: Why is the dataIndex and key given as `email`?

File: app/javascript/src/components/Dashboard/Contacts/index.jsx [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+import { Container, Header } from "neetoui/layouts";
+
+import contactsApi from "apis/contacts";
+
+import DeleteAlert from "./DeleteAlert";
+import MenuBar from "./Menubar";
+import NewContactPane from "./Pane/Create";
+import Table from "./Table";
+
+const Contact = () => {
```
Comment (yedhink) [inline app/javascript/src/components/Dashboard/Contacts/index.jsx:13]: https://academy.bigbinary.com/learn-react/pages/naming-conventions-in-react

File: app/javascript/src/components/Dashboard/Contacts/index.jsx [hunk 6]
```diff
@@ -0,0 +1,93 @@
...
+  };
+
+  const handleDelete = contactId => {
+    setSelectedContactId(contactId);
+    setShowDeleteAlert(true);
+  };
+
+  const onMenuBarToggle = () => {
+    setShowMenu(prevShowMenu => !prevShowMenu);
+  };
```
Comment (yedhink) [inline app/javascript/src/components/Dashboard/Contacts/index.jsx:45]: Do we need this to be separate function? Are we using this at more than one place? If not, I think we should just invoke this setter wherever required. The setter name is pretty self-explanatory.

File: app/javascript/src/components/constants.js [hunk 7]
```diff
@@ -43,3 +43,14 @@ export const LIST_NOTES_DUMMY_DATA = {
     },
   ],
 };
+
+export const LIST_CONTACTS_DUMMY_DATA = {
+  data: Array.from({ length: 50 }, (_, i) => ({
```
Comment (yedhink) [inline app/javascript/src/components/constants.js:48]: Nice. 👍

**Review by yedhink — CHANGES_REQUESTED** [review]
@adil62

File: app/javascript/src/components/Dashboard/Contacts/Table.jsx (hunk 4 above)
Comment (adil62) [reply app/javascript/src/components/Dashboard/Contacts/Table.jsx:58]: It should have been createdAt , will resolve these issues in the next PR.

File: app/javascript/src/components/Dashboard/Contacts/DeleteAlert.jsx (hunk 1 above)
Comment (adil62) [reply app/javascript/src/components/Dashboard/Contacts/DeleteAlert.jsx:34]: All issues has been resolved in #11

**Pull request merged**